### ISSUE TITLE
Script will not terminate when using Numba njit, numba_dpex, and MPI

### ISSUE THREAD

**Issue by JosephWick** [IntelPython/numba-dpex#1475]
Myself and @beckermr are working on an MPI-parallel code that uses numba njit and a numba_dpex kernel. We have encountered an issue where, when running our code with MPI, the script will run to the end and behave as expected but will never terminate. 

I have a short script that recreates the problem below with the following behavior:
* If only the numba njit function is called, or only the numba_dpex kernel, then the script runs and terminates normally
* If both the numba njit function and the numba_dpex kernel are called and the script is run serially, it runs and terminates normally
* If both the numba njit function and the numba_depx kernel are called and the script is run with MPI, it functions as expected but will not terminate

To run, `mpiexec -np 2 python demo.py`

```
import numpy as np
import dpnp

import numba_dpex
from numba import njit, prange


# some methods
@njit(fastmath=True, parallel=True)
def numba_cpu_njit(x, y):
    n_dim = x.shape[0]
    for i in prange(n_dim):
        y[i,:] = x[i,:]


@numba_dpex.kernel
def numba_sycl(x, y):
    start = numba_dpex.get_global_id(0)
    stride = numba_dpex.get_global_size(0)

    for i in range(start, x.shape[0], stride):
        for j in range(x.shape[1]):
            y[i,j] = 2+x[i,j]


# run some code
x = np.ones((2,1000), dtype=np.float64)
y = np.zeros((2,1000), dtype=np.float64)

numba_cpu_njit(x, y)

x_dpnp = dpnp.asarray(y.astype(np.float64), device="gpu")
y_dpnp = dpnp.zeros(y.shape, device="gpu", dtype=np.float64)

x3 = numba_sycl[numba_dpex.Range(x_dpnp.shape[0])](
            x_dpnp,
            y_dpnp
)
```

**Comment by diptorupd** [comment]
@JosephWick thank you for reporting the issue. I will investigate and update here.

What version of numba-dpex are you using? I see from your reproducer that you are not using the latest API that has been released in 0.23.

**Comment by JosephWick** [comment]
@diptorupd thanks for looking into this! I am using numba-dpex version 0.22.1

**Comment by adarshyoga** [comment]
@JosephWick and @beckermr, we are unable to reproduce the hang with numba-dpex versions 0.22.1 and 0.23.0. Can you try again with any of these versions? You can get the latest version of numba-dpex (0.23.0) from `dppy/label/dev` conda channel (use `conda install -c dppy/label/dev numba-dpex`).

numba-dpex api had changes with 0.23. I modified your reproducer slightly to get it to work with 0.23.0. Here is the reproducer with 0.23.

``` python
import numpy as np
import dpnp

import numba_dpex
from numba import njit, prange
from numba_dpex import kernel_api as kapi


# some methods
@njit(fastmath=True, parallel=True)
def numba_cpu_njit(x, y):
    n_dim = x.shape[0]
    for i in prange(n_dim):
        y[i,:] = x[i,:]


@numba_dpex.kernel
def numba_sycl(item: kapi.Item, x, y):
    start = item.get_id(0)
    stride = item.get_range(0)

    for i in range(start, x.shape[0], stride):
        for j in range(x.shape[1]):
            y[i,j] = 2+x[i,j]


# run some code
x = np.ones((2,1000), dtype=np.float64)
y = np.zeros((2,1000), dtype=np.float64)

numba_cpu_njit(x, y)

x_dpnp = dpnp.asarray(y.astype(np.float64), device="gpu")
y_dpnp = dpnp.zeros(y.shape, device="gpu", dtype=np.float64)

# x3 = numba_sycl[numba_dpex.Range(x_dpnp.shape[0])](
#             x_dpnp,
#             y_dpnp
# )

x3 = numba_dpex.call_kernel(numba_sycl, kapi.Range(x_dpnp.shape[0]), x_dpnp, y_dpnp)
```

**Comment by adarshyoga** [comment]
> @diptorupd thanks for looking into this! I am using numba-dpex version 0.22.1

@JosephWick can you join us gitter, we can chat and triage it together - https://app.gitter.im/#/room/#Data-Parallel-Python_community:gitter.im

**Comment by adarshyoga** [comment]
Issue resolved after using numba-dpex 0.23.0.
